src/h3lib/lib/h3Index.c

```diff
@@ -217,38 +223,38 @@ H3Index makeDirectChild(H3Index h, int cellNumber) {
 /**
  * cellToChildren takes the given hexagon id and generates all of the children
  * at the specified resolution storing them into the provided memory pointer.
- * It's assumed that maxCellToChildrenSize was used to determine the allocation.
+ * It's assumed that cellToChildrenSize was used to determine the allocation.
  *
  * @param h H3Index to find the children of
  * @param childRes int the child level to produce
  * @param children H3Index* the memory to store the resulting addresses in
  */
 void H3_EXPORT(cellToChildren)(H3Index h, int childRes, H3Index* children) {
-    int parentRes = H3_GET_RESOLUTION(h);
-    if (!_isValidChildRes(parentRes, childRes)) {
-        return;
-    } else if (parentRes == childRes) {
-        *children = h;
-        return;
-    }
-    int bufferSize = H3_EXPORT(maxCellToChildrenSize)(h, childRes);
-    int bufferChildStep = (bufferSize / 7);
-    int isAPentagon = H3_EXPORT(isPentagon)(h);
-    for (int i = 0; i < 7; i++) {
-        if (isAPentagon && i == K_AXES_DIGIT) {
-            H3Index* nextChild = children + bufferChildStep;
-            while (children < nextChild) {
-                *children = H3_NULL;
-                children++;
-            }
-        } else {
-            H3_EXPORT(cellToChildren)
-            (makeDirectChild(h, i), childRes, children);
-            children += bufferChildStep;
-        }
+    IterCellsChildren iter = iterInitParent(h, childRes);
+
+    for (int64_t i = 0; iter.h; i++, iterStepChild(&iter)) {
```

Comment:
Same nit as before, you can put that `iter` declaration inside of the for loop as well, though this one is pretty wild because you're also adding an index to it. Would it make sense to have a second iterator method that creates the iter object along with an index counter that you can use (`iter.h` and `iter.i`)?

Reply:
How about just:

```c
    int64_t i = 0;
    for (IterCellsChildren iter = iterInitParent(h, childRes); iter.h;
         iterStepChild(&iter)) {
        children[i] = iter.h;
        i++;
    }
```

I suppose we could add an `iter.i` to count which cell you're on, but I'm not yet sure if that's any better than letting the user handle it?

Reply:
You could also do:

```c
for (IterCellsChildren iter = iterInitParent(h, childRes), int64_t i = 0; iter.h; iterStepChild(&iter), i++) {
```

I just suggested the `iter.i` because iterators in most languages actually return `(i, val)` tuples, which *kinda* looks like a struct of `i` and `val`.

Reply:
Makes sense. I won't change it in this PR, but open to updating it in the future (before we finalize 4.0) if it seems like it'd be generally useful.